### DISCUSSION TITLE
Discounts in percent: rounding

### DISCUSSION THREAD

**Discussion by orjanskaar** [vendurehq/vendure#1537] [General]
Hello

Really love working with vendure so far.  
I'm almost complete with my integration with vendure and everyting is working grat :)

In Promotions I have a smaller "problem" regarding discounts that may already have a solution, but have not found a sloution.
Discounts in percent works fine, but there is no way to round the discount. Ex Norwegian NOK "89" becomes "71.2" with 20% discount, which is correct, but not the way i want it diplayed in the storefront. 
I can mess around with how it is displayed, but at checkout there will be a mismatch.

A feature in promotion to either floor or ceil the discounted price would have been a nice addon :)

Keep up the good work!

**Comment by michaelbromley** [comment]
Hi!

Thanks for the kind words - I'm glad you are enjoying working with Vendure!

Ah, rounding problems... I know them well 😆  OK off the top of my head I _think_ you might be able to solve this using a custom [PromotionAction](https://www.vendure.io/docs/typescript-api/promotions/promotion-action/#promotionaction). For example, here's the logic from the default promotion action that does a % discount on an item: https://github.com/vendure-ecommerce/vendure/blob/c4811d42e106aa65b4611fd9aa35d893a4c5f37a/packages/core/src/config/promotion/actions/product-percentage-discount-action.ts#L27-L32

You could create a customized version of this which applies the desired rounding to the returned value.

**Reply by orjanskaar** [reply]
Thank you so much!

Created a custom action as described in [Creating custom actions](https://www.vendure.io/docs/developer-guide/promotions/)
Everything working as expected. Thank you again for your help :)

Using @vendure/core/dist/config/promotion/actions/facet-values-percentage-discount-action.js as template: 


```javascript
async execute(ctx, orderItem, orderLine, args) {
 if (await facetValueChecker.hasFacetValues(orderLine, args.facets)) {
        const unitPrice = ctx.channel.pricesIncludeTax ? orderLine.unitPriceWithTax : orderLine.unitPrice;
        const discount = Math.ceil((unitPrice * (args.discount / 100))/100)*100;
        return -discount
    }
    return 0;
  },
```
